src/pages/ServiceDetails/ServiceInfo/ServiceInfoPods.tsx

```diff
@@ -1,66 +1,97 @@
 import * as React from 'react';
-import { Col, Row } from 'patternfly-react';
-import LocalTime from '../../../components/Time/LocalTime';
+import { Col, Row, OverlayTrigger, Tooltip } from 'patternfly-react';
 import Badge from '../../../components/Badge/Badge';
 import { Pod } from '../../../types/ServiceInfo';
 import { PfColors } from '../../../components/Pf/PfColors';
+import { PodsGroup, groupPods } from './ServiceInfoPodsGrouping';
 
 interface Props {
   pods?: Pod[];
 }
 
 class ServiceInfoPods extends React.Component<Props> {
+  groups: PodsGroup[];
+
   constructor(props: Props) {
     super(props);
+    if (props.pods) {
+      this.groups = groupPods(props.pods);
+    } else {
+      this.groups = [];
+    }
   }
 
   render() {
     return (
       <div className="card-pf">
         <Row className="row-cards-pf">
           <Col xs={12} sm={12} md={12} lg={12}>
-            {(this.props.pods || []).map((pod, u) => (
+            {(this.groups || []).map((group, u) => (
               <div className="card-pf-body" key={'pods_' + u}>
-                <h3>{pod.name}</h3>
+                <h3>
+                  {group.numberOfPods > 1 ? (
+                    <OverlayTrigger
+                      // prettier-ignore
+                      overlay={<Tooltip id={'pod_names_' + u} title="Pod Names">{group.names.join(', ')}</Tooltip>}
+                      placement="top"
+                      trigger={['hover', 'focus']}
+                    >
+                      <span>{group.commonPrefix + '... (' + group.numberOfPods + ' replicas)'}</span>
+                    </OverlayTrigger>
+                  ) : (
+                    group.commonPrefix
```

Comment:
I wonder if for UX perspective, it should also be added a "(1 replica)" msg in the single result.
I don't know, users could think that instead of a pod, it's something like a group with no pods.
I see it's redundant but I missed the "(1 replica)" here just to see the same format.

Reply:
I've no objection, @abonas or @serenamarie125 no objection too?

Reply:
To have (1 replica) for 1-pod group sounds good to me, a little bit more homogeneous :)

Reply:
done